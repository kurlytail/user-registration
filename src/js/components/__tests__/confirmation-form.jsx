import React from 'react';
import ReCAPTCHA from 'react-google-recaptcha';
import Renderer from 'react-test-renderer';
import ConfirmationForm from '../confirmation-form';

jest.mock('react-google-recaptcha');

const FIXTURES = {
    actions: {
        deleteRegistration: jest.fn()
    },

    nullState: null,

    nameEvent: {
        target: {
            value: 'test-name'
        }
    },

    passwordEvent: {
        target: {
            value: 'test-name'
        }
    },

    confirmPasswordEvent: {
        target: {
            value: 'test-name'
        }
    },

    submitEvent: {
        preventDefault: jest.fn()
    }
};

describe('# confirmation-form', () => {
    beforeEach(() => {
        Object.entries(FIXTURES.actions).forEach(([name, action]) => action.mockReset());
    });

    describe('## constructor', () => {
        it('### should create default object', () => {
            const component = Renderer.create(
                <ConfirmationForm
                    email="test-email"
                    hash="test-hash"
                    state={FIXTURES.nullState}
                    actions={FIXTURES.actions}
                />
            );
            expect(component.toJSON()).toMatchSnapshot();
        });
    });

    describe('## render', () => {
        it('### render with different states', () => {
            let component = Renderer.create(
                <ConfirmationForm
                    state={FIXTURES.nullState}
                    actions={FIXTURES.actions}
                    email="test-email"
                    hash="test-hash"
                />
            );
            expect(component.toJSON()).toMatchSnapshot();
        });

        it('### render with form methods', () => {
            let component = Renderer.create(
                <ConfirmationForm
                    state={FIXTURES.nullState}
                    actions={FIXTURES.actions}
                    email="test-email"
                    hash="test-hash"
                />
            );

            let instance = component.getInstance();

            instance.handleName(FIXTURES.passwordEvent);
            expect(component.toJSON()).toMatchSnapshot();

            instance.handleConfirmPassword(FIXTURES.confirmPasswordEvent);
            expect(component.toJSON()).toMatchSnapshot();
        });

        it('### render submit methods', () => {
            const component = Renderer.create(
                <ConfirmationForm
                    state={FIXTURES.nullState}
                    actions={FIXTURES.actions}
                    email="test-email"
                    hash="test-hash"
                />
            );
            const instance = component.getInstance();
            instance.onSubmit(FIXTURES.submitEvent);

            expect(component.toJSON()).toMatchSnapshot();
            expect(FIXTURES.actions.deleteRegistration).toBeCalled();
            expect(FIXTURES.submitEvent.preventDefault).toBeCalled();
        });
    });
});
