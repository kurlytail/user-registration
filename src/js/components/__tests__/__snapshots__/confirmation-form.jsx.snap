// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# confirmation-form ## constructor ### should create default object 1`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        className="form-control"
        id="user-registration-email"
        readonly={true}
        type="email"
        value="test-email"
      />
    </div>
    <div
      className="form-label-group"
    >
      <label>
        Name
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-name"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-label-group"
    >
      <label>
        Password
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-password"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </div>
    <div
      className="form-label-group"
    >
      <label>
        Confirm Password
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-confirmPassword"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Confirm Registration
    </button>
  </div>
</form>
`;

exports[`# confirmation-form ## render ### render submit methods 1`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        className="form-control"
        id="user-registration-email"
        readonly={true}
        type="email"
        value="test-email"
      />
    </div>
    <div
      className="form-label-group"
    >
      <label>
        Name
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-name"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-label-group"
    >
      <label>
        Password
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-password"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </div>
    <div
      className="form-label-group"
    >
      <label>
        Confirm Password
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-confirmPassword"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Confirm Registration
    </button>
  </div>
</form>
`;

exports[`# confirmation-form ## render ### render with different states 1`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        className="form-control"
        id="user-registration-email"
        readonly={true}
        type="email"
        value="test-email"
      />
    </div>
    <div
      className="form-label-group"
    >
      <label>
        Name
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-name"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-label-group"
    >
      <label>
        Password
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-password"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </div>
    <div
      className="form-label-group"
    >
      <label>
        Confirm Password
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-confirmPassword"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Confirm Registration
    </button>
  </div>
</form>
`;

exports[`# confirmation-form ## render ### render with form methods 1`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        className="form-control"
        id="user-registration-email"
        readonly={true}
        type="email"
        value="test-email"
      />
    </div>
    <div
      className="form-label-group"
    >
      <label>
        Name
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-name"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-label-group"
    >
      <label>
        Password
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-password"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </div>
    <div
      className="form-label-group"
    >
      <label>
        Confirm Password
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-confirmPassword"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Confirm Registration
    </button>
  </div>
</form>
`;

exports[`# confirmation-form ## render ### render with form methods 2`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        className="form-control"
        id="user-registration-email"
        readonly={true}
        type="email"
        value="test-email"
      />
    </div>
    <div
      className="form-label-group"
    >
      <label>
        Name
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-name"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-label-group"
    >
      <label>
        Password
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-password"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </div>
    <div
      className="form-label-group"
    >
      <label>
        Confirm Password
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-confirmPassword"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Confirm Registration
    </button>
  </div>
</form>
`;
